### DISCUSSION TITLE
Custom FlxTilemap tile

### DISCUSSION THREAD

**Discussion by bogdan-ov** [HaxeFlixel/flixel#2629] [Q&A] [answered]
I want to create my own tile class for FlxTilemap

I created my tilemap by extending FlxTilemap (everything works fine)
```haxe
class MyTilemap extends FlxTilemap {
// ...
}
```

I found this part of code in flixel. I thought at first to extend FlxBaseTilemap (like `class MyTilemap extends FlxBaseTilemap<MyTile>`) for my tilemap but I found that there is a lot of code in the FlxTilemap class, and copy & paste all this code isn't the solution
```haxe
class FlxTilemap extends FlxBaseTilemap<FlxTile> {
// ...
}
```

**Thanks!**

**Comment by Geokureli** [comment, marked as answer]
Im not sure if there is a better solution, currently.

Id have to look to make sure, but it might be a good idea to change haxeFlixel to make this easier. Maybe change `FlxTilemap` into `FlxTypedTilemap<T:FlxTile>`

And then add `typedef FlxTilemap = FlxTypedTilemap<FlxTile>;`

**Reply by bogdan-ov** [reply]
I also wanted to do this, but I thought it was bad because if I want to update flixel, all my changes will disappear. But most likely there is no other way...
Thanks!

**Reply by Geokureli** [reply]
If it works make a pr to haxe flixel
